docarray/array/mixins/find.py

```diff
@@ -124,14 +127,10 @@ def find(
 
         from ... import Document, DocumentArray
 
-        if isinstance(query, dict):
+        if isinstance(query, dict) and filter is None:
             return self._filter(query)
-        elif isinstance(query, (DocumentArray, Document)):
-
-            if isinstance(query, Document):
-                query = DocumentArray(query)
-
-            _query = query.embeddings
+        elif query is None and isinstance(filter, dict):
+            return self._filter(filter)
```

Comment:
in filter tests, we need to also test the API:
`da.find(filter=filter)` not just `da.find(filter)`

Reply:
done in `aacf3fe6637582c3fc4a0749b60a934bb2256b07`